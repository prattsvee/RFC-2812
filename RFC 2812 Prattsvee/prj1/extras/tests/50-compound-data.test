%{ [:a, 22] => { [1, 2, 3], :x },
   x: [99, %{ a: 33 }]
}

{ [1, 2], {:a, 22}, %{ a: 99, :b => 11} }

[ {1, 2}, %{[:x] => 33, b: 44}, :c, [], [:d, 55] ]
