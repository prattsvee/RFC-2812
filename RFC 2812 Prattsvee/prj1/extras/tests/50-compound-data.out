[
  {
    "%k": "map",
    "%v": [
      [
        {
          "%k": "list",
          "%v": [
            {
              "%k": "atom",
              "%v": ":a"
            },
            {
              "%k": "int",
              "%v": 22
            }
          ]
        },
        {
          "%k": "tuple",
          "%v": [
            {
              "%k": "list",
              "%v": [
                {
                  "%k": "int",
                  "%v": 1
                },
                {
                  "%k": "int",
                  "%v": 2
                },
                {
                  "%k": "int",
                  "%v": 3
                }
              ]
            },
            {
              "%k": "atom",
              "%v": ":x"
            }
          ]
        }
      ],
      [
        {
          "%k": "atom",
          "%v": ":x"
        },
        {
          "%k": "list",
          "%v": [
            {
              "%k": "int",
              "%v": 99
            },
            {
              "%k": "map",
              "%v": [
                [
                  {
                    "%k": "atom",
                    "%v": ":a"
                  },
                  {
                    "%k": "int",
                    "%v": 33
                  }
                ]
              ]
            }
          ]
        }
      ]
    ]
  },
  {
    "%k": "tuple",
    "%v": [
      {
        "%k": "list",
        "%v": [
          {
            "%k": "int",
            "%v": 1
          },
          {
            "%k": "int",
            "%v": 2
          }
        ]
      },
      {
        "%k": "tuple",
        "%v": [
          {
            "%k": "atom",
            "%v": ":a"
          },
          {
            "%k": "int",
            "%v": 22
          }
        ]
      },
      {
        "%k": "map",
        "%v": [
          [
            {
              "%k": "atom",
              "%v": ":a"
            },
            {
              "%k": "int",
              "%v": 99
            }
          ],
          [
            {
              "%k": "atom",
              "%v": ":b"
            },
            {
              "%k": "int",
              "%v": 11
            }
          ]
        ]
      }
    ]
  },
  {
    "%k": "list",
    "%v": [
      {
        "%k": "tuple",
        "%v": [
          {
            "%k": "int",
            "%v": 1
          },
          {
            "%k": "int",
            "%v": 2
          }
        ]
      },
      {
        "%k": "map",
        "%v": [
          [
            {
              "%k": "list",
              "%v": [
                {
                  "%k": "atom",
                  "%v": ":x"
                }
              ]
            },
            {
              "%k": "int",
              "%v": 33
            }
          ],
          [
            {
              "%k": "atom",
              "%v": ":b"
            },
            {
              "%k": "int",
              "%v": 44
            }
          ]
        ]
      },
      {
        "%k": "atom",
        "%v": ":c"
      },
      {
        "%k": "list",
        "%v": []
      },
      {
        "%k": "list",
        "%v": [
          {
            "%k": "atom",
            "%v": ":d"
          },
          {
            "%k": "int",
            "%v": 55
          }
        ]
      }
    ]
  }
]
